pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -546,11 +551,14 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 
 					// Already got a list of endpoints to watch and it is the same as the request, this is an ack
 					if reflect.DeepEqual(con.Clusters, clusters) {
-						if discReq.ErrorDetail == nil && discReq.ResponseNonce != "" {
+						adsLog.Debugf("ADS:EDS: ACK %s %s (%s) %s %s", peerAddr, con.ConID, con.modelNode, discReq.VersionInfo, discReq.ResponseNonce)
+						if discReq.ResponseNonce != "" {
```

Comment:
What happened with ErrorDetail==nil ? Nacks are not acks

Reply:
I think you mean What happened with ErrorDetail!=nil ? 

It is checked at first in L530 if ErrorDetail!=nil continue skip the following.